### PR TITLE
FieldConfig: Unify the custom and standard registry

### PR DESCRIPTION
**Pull request by torkelo**
This unifies the registries and builds a single registry per panel plugin. 

* Had to introduce a path property to make ids not conflict (so we can prefix custom options with custom). Another option could be to use lodash path set to be able to set directly properties on `custom.width` for example.

Todo
- [x] Update all code path so this actually works for both standard and custom (had it working for a while with standard and custom showing, but custom not actually working/updating)
- [ ] Update tests

### CODE REVIEW

File: public/app/plugins/panel/table2/module.tsx [hunk 1]
```diff
@@ -1,39 +1,44 @@
...
+            options: [
+              { value: 'auto', label: 'Auto' },
+              { value: 'color-background', label: 'Color background' },
+              { value: 'gradient-gauge', label: 'Gradient gauge' },
+              { value: 'lcd-gauge', label: 'LCD gauge' },
+            ],
+          },
+        });
+    },
   })
```
Comment (torkelo) [inline public/app/plugins/panel/table2/module.tsx:38]: This is the main new api to set custom field options, and optionally set order & defaults and what options to show 
@ryantxu

Comment (torkelo) [reply public/app/plugins/panel/table2/module.tsx:38]: all properties are optional in this call so if you do not pass anything of if any property is not specified defaults will be used

**Comment by ryantxu** [comment]
I like it... this is good: 
image

---

Minor nit/alternative... maybe rather than:
```
useFieldConfigOptions
...
useCustomOptions: builder => {
```
we want:
```
useFieldConfig
...
useCustomConfig: builder => {
```

The plugin is using *config* and the config has the options.  kinda minor but at least it is shorter ;)

**Comment by torkelo** [comment]
> useFieldConfig
...
useCustomConfig: builder => {

👍 yea, like that

**Comment by torkelo** [comment]
if you get merge conflicts in table2/module.tsx it's was just me removing the defaultValue 1 for custom.width property (was causing issues with table panel)

**Pull request merged**